File: yilu_an_backend/app/api/v1/voice_log.py
from fastapi import APIRouter, Depends, HTTPException, status
from typing import List
from datetime import datetime
from app.schemas.voice_log import VoiceLogCreate, VoiceLogUpdate, VoiceLogResponse
from app.services.voice_log import VoiceLogService
from app.dependencies import get_voice_log_service, get_current_active_user
from app.models import User

router = APIRouter()

@router.post("/", response_model=VoiceLogResponse, status_code=status.HTTP_201_CREATED)
async def create_voice_log(
    log: VoiceLogCreate,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    return voice_log_service.create_log(log)

@router.get("/", response_model=List[VoiceLogResponse])
async def get_voice_logs(
    user_id: int,
    limit: int = 100,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    return voice_log_service.get_logs_by_user_id(user_id, limit)

@router.get("/record/{record_id}", response_model=List[VoiceLogResponse])
async def get_voice_logs_by_record(
    record_id: int,
    limit: int = 100,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    return voice_log_service.get_logs_by_record_id(record_id, limit)

@router.get("/time-range/{user_id}", response_model=List[VoiceLogResponse])
async def get_voice_logs_by_time_range(
    user_id: int,
    start_time: datetime,
    end_time: datetime,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    return voice_log_service.get_logs_by_time_range(user_id, start_time, end_time)

@router.get("/recent/{user_id}", response_model=List[VoiceLogResponse])
async def get_recent_voice_logs(
    user_id: int,
    hours: int = 24,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    return voice_log_service.get_recent_logs(user_id, hours)

@router.get("/{log_id}", response_model=VoiceLogResponse)
async def get_voice_log(
    log_id: int,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    log = voice_log_service.get_log_by_id(log_id)
    if not log:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="语音日志不存在"
        )
    return log

@router.put("/{log_id}", response_model=VoiceLogResponse)
async def update_voice_log(
    log_id: int,
    log: VoiceLogUpdate,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    updated_log = voice_log_service.update_log(log_id, log)
    if not updated_log:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="语音日志不存在"
        )
    return updated_log

@router.delete("/{log_id}", status_code=status.HTTP_204_NO_CONTENT)
async def delete_voice_log(
    log_id: int,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    success = voice_log_service.delete_log(log_id)
    if not success:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail="语音日志不存在"
        )

@router.delete("/user/{user_id}/old", response_model=dict)
async def delete_old_voice_logs(
    user_id: int,
    days: int = 30,
    voice_log_service: VoiceLogService = Depends(get_voice_log_service),
    current_user: User = Depends(get_current_active_user)
):
    deleted_count = voice_log_service.delete_old_logs(user_id, days)
    return {"deleted_count": deleted_count}

File: yilu_an_frontend/miniprogram/api/voice-log.ts
// 语音日志相关接口
import { api } from '../utils/request';

// 创建语音日志请求参数
export interface CreateVoiceLogParams {
  user_id: number;
  device_id: number;
  voice_text: string;
  audio_url?: string;
  duration?: number;
}

// 更新语音日志请求参数
export interface UpdateVoiceLogParams {
  voice_text?: string;
  audio_url?: string;
  duration?: number;
}

// 语音日志
export interface VoiceLog {
  id: number;
  user_id: number;
  device_id: number;
  voice_text: string;
  audio_url: string | null;
  duration: number | null;
  created_at: string;
  updated_at: string;
}

// 语音日志相关API
export const voiceLogApi = {
  // 创建语音日志
  createLog: (params: CreateVoiceLogParams) => {
    return api.post<VoiceLog>('/voice-logs/', params);
  },
  
  // 获取用户的语音日志列表
  getUserLogs: (userId: number, limit?: number) => {
    return api.get<VoiceLog[]>('/voice-logs/', { data: { user_id: userId, limit } });
  },
  
  // 根据设备ID获取语音日志列表
  getDeviceLogs: (deviceId: number, limit?: number) => {
    return api.get<VoiceLog[]>(`/voice-logs/device/${deviceId}`, { data: { limit } });
  },
  
  // 根据时间范围获取语音日志
  getLogsByTimeRange: (userId: number, startTime: string, endTime: string) => {
    return api.get<VoiceLog[]>(`/voice-logs/time-range/${userId}`, {
      data: { start_time: startTime, end_time: endTime }
    });
  },
  
  // 获取最近几小时的语音日志
  getRecentLogs: (userId: number, hours?: number) => {
    return api.get<VoiceLog[]>(`/voice-logs/recent/${userId}`, { data: { hours } });
  },
  
  // 根据ID获取语音日志
  getLogById: (logId: number) => {
    return api.get<VoiceLog>(`/voice-logs/${logId}`);
  },
  
  // 更新语音日志
  updateLog: (logId: number, params: UpdateVoiceLogParams) => {
    return api.put<VoiceLog>(`/voice-logs/${logId}`, params);
  },
  
  // 删除语音日志
  deleteLog: (logId: number) => {
    return api.delete(`/voice-logs/${logId}`);
  },
};
